ql/time/schedule.cpp

```diff
@@ -463,6 +463,39 @@ namespace QuantLib {
 
     }
 
+    Schedule Schedule::after(const Date& truncationDate) const {
+        Schedule result = *this;
+
+        QL_REQUIRE(truncationDate < result.dates_.back(),
+            "truncation date " << truncationDate <<
+            " must be before the last schedule date " <<
+            result.dates_.back());
+        if (truncationDate > result.dates_[0]) {
+            // remove earlier dates
+            while (result.dates_[0] < truncationDate) {
+                result.dates_.erase(result.dates_.begin());
+                if (!result.isRegular_.empty())
+                    result.isRegular_.erase(result.isRegular_.begin());
+            }
+
+            // add truncationDate if missing
+            if (truncationDate != result.dates_.front()) {
+                result.dates_.insert(result.dates_.begin(), truncationDate);
+                result.isRegular_.insert(result.isRegular_.begin(), false);
+                result.terminationDateConvention_ = Unadjusted;
+            }
+            else {
+                result.terminationDateConvention_ = convention_;
+            }
+
+            if (result.nextToLastDate_ <= truncationDate)
```

Comment:
To be honest, I'm not sure what this block is meant to achieve. It appears in `Schedule::until`, so I copied it and change the `>=` to `<=`.

Reply:
If either stub date is outside the new domain, this block removes stored information about them from the new schedule.